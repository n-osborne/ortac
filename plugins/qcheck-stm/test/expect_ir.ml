open Ortac_qcheck_stm

let () =
  let go v =
    let open Ir in
    let str =
      "Value "
      ^ v.id
      ^ " of type "
      ^ Ppxlib.string_of_core_type v.ty
      ^ " will be tested."
    in
    print_endline str
  in
  match Config.init "lib.mli" "good_init" "sut" with
  | Error err -> print_endline err
  | Ok (config, sigs) -> Ir_of_gospel.signature config sigs |> List.iter go
