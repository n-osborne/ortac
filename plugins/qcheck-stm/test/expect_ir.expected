Value f of type sut -> int will be tested.
